spec/graphql/graphql-hive/analyzer_spec.rb

```diff
@@ -349,4 +349,35 @@
       )
     end
   end
+
+  context "with an invalid field" do
+    let(:query_string) do
+      %|
+      query getGatewayProjects {
+        projectsByManyTypes(type: [STITCHING]),
+        nonExistentField {
+          subField
+        }
+      }
+      |
+    end
+
+    it "collects all valid fields and excludes invalid fields" do
+      expect(used_fields).to include(
+        "Query",
+        "Query.projectsByManyTypes",
+        "ProjectType",
+        "Query.projectsByManyTypes.type",
+        "ProjectType.STITCHING"
+      )
+      expect(used_fields).not_to include(
+        "Query.nonExistentField",
+        "nonExistentField.subField"
+      )
+    end
+
+    it "does not raise an error when encountering an invalid field" do
+      expect { used_fields }.not_to raise_error
+    end
```

Comment:
For situations like this `contain_exactly` is preferable. With `include` it is possible for `used_fields` to include data we are not accounting for. We should be precise in expect statements to prevent future false positives. 

```suggestion
    it "collects all valid fields and excludes invalid fields" do
      expect(used_fields).to contain_exactly(
        "Query",
        "Query.projectsByManyTypes",
        "ProjectType",
        "Query.projectsByManyTypes.type",
        "ProjectType.STITCHING"
      )

    it "does not raise an error when encountering an invalid field" do
      expect { used_fields }.not_to raise_error
    end
```

Reply:
using `contain_exactly` makes sense in this scenario! i updated the test and removed the second one [here](https://github.com/rperryng/graphql-ruby-hive/pull/44/commits/eefd2ed36bd5614550465e796872b36a26df7a13)